newdle/client/src/components/answer/AnswerPage.js

```diff
@@ -86,19 +101,25 @@ export default function AnswerPage() {
         ({code}) => client.updateParticipantAnswers(newdle.code, code, availabilityData)
       );
 
-  const canSubmit = (participantCode || name.length >= 2) && !submitting;
+  const canSubmit = (participantCode || user || name.length >= 2) && !submitting;
 
   async function answerNewdle() {
     await submitAnswer(newdle.code, participantCode || name, availabilityData);
     setSaved(true);
   }
 
   useEffect(() => {
-    if (newdle && participantCode) {
-      dispatch(fetchParticipant(newdle.code, participantCode));
+    if (newdle && (participantCode || user)) {
+      dispatch(fetchParticipant(newdle.code, participantCode, !!user));
     }
   }, [newdle, participantCode, dispatch]);
 
+  useEffect(() => {
+    if (user && !participantCode && participant) {
+      history.push(`/newdle/${newdle.code}/${participant.code}`);
```

Comment:
i'd probably use `.replace()` here. pushing something where using the back button will immediately redirect you again is usually not great because it pretty much breaks the back button